source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -244,6 +244,24 @@ ShaderPtr OslShaderGenerator::generate(const string& name, ElementPtr element, G
     _syntax->makeIdentifier(functionName, graph.getIdentifierMap());
     setFunctionName(functionName, stage);
     emitLine(functionName, stage, false);
+
+    // Add any metadata if set on the graph.
+    const ShaderMetadataVecPtr& metadata = graph.getMetadata();
+    if (metadata && metadata->size())
+    {
+        emitScopeBegin(stage, Syntax::DOUBLE_SQUARE_BRACKETS);
+        for (size_t j = 0; j < metadata->size(); ++j)
+        {
+            const ShaderMetadata& data = metadata->at(j);
+            const string& delim = (j == metadata->size() - 1) ? EMPTY_STRING : COMMA;
+            const string& dataType = _syntax->getTypeName(data.type);
+            const string dataValue = _syntax->getValue(data.type, *data.value, true);
+            emitLine(dataType + " " + data.name + " = " + dataValue + delim, stage, false);
```

Comment:
I don't recall. Do you need to add quotes to string data values, or will data value contain quotes ?

Reply:
In OSL metadata declarations you use quotes only for string values. For example:

```
    color base_color = color(1, 1, 1)
    [[
        color min = color(0, 0, 0),
        color max = color(1, 1, 1),
        string label = "Base Color",
        string page = "Base",
        string help = "Color of the diffuse reflection."
    ]],
```

Reply:
The call to `_syntax->getValue(...)` above will make sure the value is in the right syntax.